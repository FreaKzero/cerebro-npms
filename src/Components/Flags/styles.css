.flags {

}

.flagInsecure {
  padding: 0 3px 0 3px;
  border-radius: 2px;
  border: none;
  color: white;
  font-size: 10px;
  margin-right: 5px;
  background-color: #BE1C1C;
}

.flagDeprecated {
  padding: 0 3px 0 3px;
  border-radius: 2px;
  border: none;
  color: white;
  font-size: 10px;
  margin-right: 5px;
  background-color: #BE1C1C;
}

.flagUnstable {
  padding: 0 3px 0 3px;
  border-radius: 2px;
  border: none;
  color: white;
  font-size: 10px;
  margin-right: 5px;
  background-color: #616161;
}

.big .flagInsecure {
  font-size: 12px;
  padding: 0 5px 0 5px;
  font-weight: normal;
}

.big .flagDeprecated {
  font-size: 12px;
  padding: 0 5px 0 5px;
  font-weight: normal;
}

.big .flagUnstable {
  font-size: 12px;
  padding: 0 5px 0 5px;
  font-weight: normal;
}