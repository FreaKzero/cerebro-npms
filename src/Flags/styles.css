.flags {
  margin-top: 5px;
}

.flag {
  padding: 0 3px 0 3px;
  border-radius: 2px;
  border: none;
  background-color: #BE1C1C;
  color: white;
  font-size: 10px;
  margin-right: 5px;
}