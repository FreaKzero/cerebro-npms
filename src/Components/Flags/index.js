import React, { Component, PropTypes } from 'react';
import styles from './styles.css'

export default ({flags, size}) => (
  <span className={size === 'big' ? styles.big : styles.small}>
    <span className={styles.flags}>
      {flags.deprecated ? <span className={styles.flagDeprecated}>Deprecated</span> : null}
      {flags.insecure ? <span className={styles.flagInsecure}>insecure</span> : null}
      {flags.unstable ? <span className={styles.flagUnstable}>unstable</span> : null}
    </span>
  </span>
)