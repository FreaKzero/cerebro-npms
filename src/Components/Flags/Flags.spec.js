import React from 'react';
import Flags from './index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {   

  const flags = {
    flags: {
      deprecated: true
    }
  }
  
    const actions = {
      open: () => { return; },
      hideWindow: () => { return; }
    }

  const tree = renderer
    .create(<Flags flags={flags} size='big' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});