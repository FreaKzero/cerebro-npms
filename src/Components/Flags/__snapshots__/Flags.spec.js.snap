// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className={undefined}
>
  <span
    className={undefined}
  />
</span>
`;
